USING_DOCKER.md

```diff
@@ -0,0 +1,123 @@
+# Using the Dockerfile
+
+The `Dockerfile` in this repo builds a Docker container
+that has current versions of all the development tools.
+Using Docker means you do not have to concern
+yourself that these tools will conflict with
+other software on your computer.
+
+## Development cycle
+
+The developer loop when using Docker is substantially the same as
+if the tools had been installed directly.
+
+All the source files live in the `prql` directory on your machine.
+Just use `git` to pull or to push the `prql` directory as normal.
+As you edit the source, the tools (wrapped in the Docker container)
+watch those directories and re-run
+so you can see your results instantly.
+
+When you exit the Docker container (say, at the end of the development
+session), the `prql` directory on the local machine contains the
+latest files. 
+
+To do all this, build the Docker container and start it
+as described in the **Installation** section.
+Then read the separate steps in **Running components under Docker**
+for each component you wish to work on.
+
+## Installation
+
+First install Docker on your computer,
+using one of the many guides on the internet.
+
+Next build the Docker container with the following commands.
+_(It will take some time while Docker pulls in all the
+necessary developer tools.)_
+
+```bash
+cd <top-level-PRQL-directory>
+docker build -t prql .
+```
+
+Finally, start up the Docker container with:
+
+```bash
+cd <top-level-PRQL-directory>
+docker run -it -v $(pwd)/:/app -p 3000:3000 prql
+```
+
+- You'll see a `root@xxxxxxxxx:/app/#` prompt
+- Enter the commands below for the component you're working on
+- Ctl-C to exit that component
+- Enter 'exit' to close down the Docker machine
+
+## Running components under Docker
+
+The following commands come from the `README.md`
+file in each of the component directories.
+Use the `docker run...` command above, then enter
+the command(s) below.
+
+_(The first time you run a component, it may take some
+time to install additional files.
+Once they're built, start up is quick.)_
+
+**Playground:** Use the command above, then enter:
+
+```bash
+cd playground
+npm install # first time only
+npm start
+```
+
+**Language Book:** Use the command above, then enter:
+
+```
+cd book
+mdbook serve -n 0.0.0.0 -p 3000
```

Comment:
Note for the project – we could add this & the hugo command as `task`s, or add configs so we don't need to pass args to them

Reply:
Yes - but one insurmountable hurdle at a time... :-)